src/uri/path.rs

```diff
@@ -49,6 +49,11 @@ impl PathAndQuery {
             match URI_CHARS[b as usize] {
                 0 => {
                     if b == b'%' {
+                        // Check if next character is not %
+                        if b'%' == src[i + 1] {
```

Comment:
I believe this will panic if `%` were the last character, indexing out of bounds.

Reply:
classic, thanks for noticing! fixed and added additional test